src/core/tools/hpc/hpc_network_provider.bsd.cpp

```diff
@@ -241,11 +241,17 @@ namespace dsn
                 {
                     message_ex* msg = _parser->get_message_on_receive(sz, read_next);
 
-                    while (msg != nullptr)
+                    while (read_next != -1 && msg != nullptr)
```

Comment:
That's right.  Comments should be add in the base class of "message_parser", and this is my fault.